src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -546,20 +546,25 @@ private void ReportUseSiteDiagnosticForDynamic(DiagnosticBag diagnostics, Identi
 
             if (node.IsTypeInContextWhichNeedsDynamicAttribute())
             {
-                if ((object)this.Compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctor) == null ||
-                    (object)this.Compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctorTransformFlags) == null)
-                {
-                    // CONSIDER:    Native compiler reports error CS1980 for each syntax node which binds to dynamic type, we do the same by reporting a diagnostic here.
-                    //              However, this means we generate multiple duplicate diagnostics, when a single one would suffice.
-                    //              We may want to consider adding an "Unreported" flag to the DynamicTypeSymbol to suppress duplicate CS1980.
+                ReportUseSiteDiagnosticForDynamic(diagnostics, node.Location);
+            }
+        }
 
-                    // CS1980: Cannot define a class or member that utilizes 'dynamic' because the compiler required type '{0}' cannot be found. Are you missing a reference?
-                    var info = new CSDiagnosticInfo(ErrorCode.ERR_DynamicAttributeMissing, AttributeDescription.DynamicAttribute.FullName);
-                    Symbol.ReportUseSiteDiagnostic(info, diagnostics, node.Location);
-                }
+        public void ReportUseSiteDiagnosticForDynamic(DiagnosticBag diagnostics, Location location)
```

Comment:
I believe there is an advantage in having an API that takes a node rather than a location. It allows to avoid unnecessary allocation in non-error scenario.